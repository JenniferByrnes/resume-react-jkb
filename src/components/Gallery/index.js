import React from 'react'
import { capitalizeFirstLetter } from '../../utils/helpers';
import PhotoList from '../PhotoList'

function Gallery({ currentCategory }) {
  const {
    name,
    description
  } = currentCategory;
  return (
    <section >
      <h1 data-testid="h1tag">{capitalizeFirstLetter(name)}</h1>
      <p>{description}</p>
      <PhotoList category={currentCategory.name} />
    </section>
  )
}

export default Gallery;