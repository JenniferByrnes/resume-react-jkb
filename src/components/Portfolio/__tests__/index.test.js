import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Gallery from '..'

const portrait = { name: "portraits", description: "Portraits of people in my life" };

afterEach(cleanup)

describe('Gallery is rendering', () => {

  it('renders', () => {
    render(<Gallery currentCategory={portrait} />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Gallery currentCategory={portrait} />)
    expect(asFragment()).toMatchSnapshot()
  })
})