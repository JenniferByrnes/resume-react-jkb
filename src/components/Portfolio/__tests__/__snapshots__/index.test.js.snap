// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery is rendering matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Portraits
    </h1>
    <p>
      Portraits of people in my life
    </p>
    <div>
      <div
        class="flex-row"
      >
        <img
          alt="Cat green eyes"
          class="img-thumbnail mx-1"
          src="0.jpg"
        />
        <img
          alt="Green parrot"
          class="img-thumbnail mx-1"
          src="1.jpg"
        />
        <img
          alt="Yellow macaw"
          class="img-thumbnail mx-1"
          src="2.jpg"
        />
        <img
          alt="Pug smile"
          class="img-thumbnail mx-1"
          src="3.jpg"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
